x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/syncjob/action/ListConnectorSyncJobsAction.java

```diff
@@ -90,27 +105,30 @@ public boolean equals(Object o) {
             Request request = (Request) o;
             return Objects.equals(pageParams, request.pageParams)
                 && Objects.equals(connectorId, request.connectorId)
-                && connectorSyncStatus == request.connectorSyncStatus;
+                && connectorSyncStatus == request.connectorSyncStatus
+                && connectorSyncJobTypeList == request.connectorSyncJobTypeList;
         }
 
         @Override
         public int hashCode() {
-            return Objects.hash(pageParams, connectorId, connectorSyncStatus);
+            return Objects.hash(pageParams, connectorId, connectorSyncStatus, connectorSyncJobTypeList);
         }
 
         private static final ConstructingObjectParser<ListConnectorSyncJobsAction.Request, String> PARSER = new ConstructingObjectParser<>(
             "list_connector_sync_jobs_request",
             p -> new ListConnectorSyncJobsAction.Request(
                 (PageParams) p[0],
                 (String) p[1],
-                p[2] != null ? ConnectorSyncStatus.fromString((String) p[2]) : null
+                p[2] != null ? ConnectorSyncStatus.fromString((String) p[2]) : null,
+                p[3] != null ? Arrays.stream(((String) p[3]).split(",")).map(ConnectorSyncJobType::fromString).toList() : null
             )
         );
 
         static {
             PARSER.declareObject(constructorArg(), (p, c) -> PageParams.fromXContent(p), PAGE_PARAMS_FIELD);
             PARSER.declareString(optionalConstructorArg(), CONNECTOR_ID_FIELD);
             PARSER.declareString(optionalConstructorArg(), ConnectorSyncJob.STATUS_FIELD);
+            PARSER.declareString(optionalConstructorArg(), ConnectorSyncJob.JOB_TYPE_FIELD);
```

Comment:
Given that we pass sync job types and statuses as a part of URL params, why do we need to add them to body parser? Can we avoid defining those fields here?

Reply:
I see `CONNECTOR_ID_FIELD` and `ConnectorSyncJob.STATUS_FIELD` are defined here, and I'm adding a new field that's why I added it here. You also mentioned in slack:
>The parser in the Request class is used to convert user provided JSON payload (from request body) into a java class

And I also see every failed tests in `ListConnectorSyncJobsActionRequestBWCSerializingTests`. Will investigate further.

Reply:
this is causing the issue you should change the parser to declare JOB_TYPE_FIELD as `PARSER.declareStringList` (or sth like that, browser available methods, there is def predefined one) 

note that in the toXContent you write an array to `JOB_TYPE_FIELD` field

Reply:
Fix the test: https://github.com/elastic/elasticsearch/pull/104855/commits/37f5e01fe5a6ed9f6c3bea0a5a62eb021769ffeb